src/core/model/Stroke.h

```diff
@@ -20,6 +22,18 @@ enum StrokeTool { STROKE_TOOL_PEN, STROKE_TOOL_ERASER, STROKE_TOOL_HIGHLIGHTER }
 enum StrokeCapStyle { ROUND, BUTT, SQUARE };
 
 class ErasableStroke;
+struct PaddedBox;
+struct PathParameter;
+
+template <class T, unsigned int N>
+class TinyVector;
```

Comment:
I think, the dependency is pulled in somewhere else. I would include it directly. Otherwise, the compilation will fail, if the indirect dep is not included there anymore.